### DISCUSSION TITLE
Add a short guide on how to resolve memory induced reboots on Creality K1 series machines

### DISCUSSION THREAD

**Discussion by DrywFiltiarn** [Guilouz/Creality-Helper-Script-Wiki#699] [Features Request]
The Wiki page on the Advanced Input Shapers already mentions that a Klipper reset can occur when the process needs more memory than can be provided by the machines mainboard, however it only suggests just "rebooting" which in my experience (and I tried several times) doesn't change much at all, since the same services that block memory will be started again.

What helped me to prevent the resets (both occuring in trying to run the shaper graphs as well as the actual calibration process) is to manually kill some services from the SSH terminal into the machine, which as we need to be rooted anyway is an easy process.

What released sufficient memory for me to be successfully run all the AIS features without crashing was to kill -KILL the following processes:
* octoeverywhere
* mobileraker
* mjpg_streamer

Results may vary depending on what people have installed obviously. But since the 3 of them aren't absolutely required to be able to do a present-at-machine shaper test/calibration they can be shut down without any trouble. Just killing these 3 processes on the machine resulted in a reduction of memory load from 57% to 36%, which apparently was more then enough to be able to run de AIS features.

![image](https://github.com/user-attachments/assets/fcdabc5f-1eac-42fe-816e-24c216eb9813)

**Comment by EchotreePrints** [comment]
If Possible, I'd like to see the advanced shaper stop the services itself before running the calibration/calculations. 
with klipper, it should be possible to execute shell commands from gcode.

**Reply by DrywFiltiarn** [reply]
That would even be better indeed as that also allows for automatically restarting those services after the calibrations have been completed.

Think it's definitely worth investigating the options since I've seen plenty people who have been running into it, certainly when people go overboard on installing additional services and tools.
